staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -770,6 +778,51 @@ func (vs *VSphere) InstanceShutdownByProviderID(ctx context.Context, providerID
 	return !isActive, nil
 }
 
+// InstanceMetadataByProviderID returns metadata of the specified instance.
+func (vs *VSphere) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
+	if providerID == "" {
+		return nil, fmt.Errorf("couldn't compute InstanceMetadata for empty providerID")
+	}
+
+	nodeName, err := vs.GetNodeNameFromProviderID(providerID)
```

Comment:
It seems NodeAddressesByProviderID is converting it differently:
https://github.com/kubernetes/kubernetes/blob/0f2cccc98c3bf2a512ec55cbeba89dbfe7c768cf/staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go#L707

Reply:
It's a logic from InstanceExistsByProviderID too. It will be cleaner to use providerID as nodeName directly if it won't cause any issue(I'm not sure now).